### ISSUE TITLE
false positive

### ISSUE THREAD

**Issue by martimlobao** [PyCQA/pylint#5784]
### Bug description

Just tried this out with the latest version of the main branch (any version after bfeca4c) and there seems to be a case where #5547 is still raising a false positive unexpected-keyword-arg error.

```python
def example_decorator(*new_func_args, **new_func_kwargs):
    def decorator(func):
        def wrapper(*args, **kwargs):
            print_me = kwargs.pop("print_me", None)
            def inner(*inner_args, **inner_kwargs):
                print(print_me)
                return func(*inner_args, **inner_kwargs)

            return inner(*args, **kwargs)

        return wrapper

    if len(new_func_args) == 1 and len(new_func_kwargs) == 0 and callable(new_func_args[0]):
        return decorator(new_func_args[0])
    return decorator

@example_decorator
def example_func(num):
    return num ** 2

example_func(6, print_me="hello")
```

(Note that no error is raised by pylint if we remove the top-level function (`example_decorator`). The reason for someone to define the decorator this way is so that it's possible to use the bare `@example_decorator`, as well as with arguments `@example_decorator(foo=1)`.)

Using pylint on commit hash e75e37a2908ef1c003ea5816c1e234523bdd38be, this is the error that's printed out:

```
E1123: Unexpected keyword argument 'print_me' in function call (unexpected-keyword-arg)
```

Calling `example_func` works fine:

```
>>> example_task(6, print_me="hello")
hello
36
```

### Command used

```shell
pylint a.py
```

### Pylint output

```shell
************* Module a
E1123: Unexpected keyword argument 'print_me' in function call (unexpected-keyword-arg)
```

### Expected behavior

No `unexpected-keyword-arg` error raised

### Pylint version

```shell
current pylint version in the `main` branch (set for 2.13.0)
```

**Comment by rmorshea** [comment]
Unsure if this is a true solution, but perhaps related: https://github.com/PyCQA/pylint/pull/8332

**Comment by Archmonger** [comment]
All potentially related issues:
- #7643
- #7437
- #8303
- #5735
- #3957
- #3954
- #3662
- #3641
- #5784
- #2578
- #1694

**Comment by mamiu** [comment]
Have the same problem. Here's a very simplified version of the code that displays this error:

```python
def clean_html(html):
    elements_to_delete = [
        {"search_term": "footer", "delete_after": True}
    ]

    for etd in elements_to_delete:
        search_term = etd.pop("search_term", None)
        elements = html.find_all(search_term)

        html = delete_elements(html, elements, **etd)

        # ... rest of the code
```

This is the error I get:

```python
        html = delete_elements(html, elements, **etd)
#              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
#              Pylint: Unexpected keyword argument 'search_term' in function call
```

BTW: This is the function definition of the `delete_elements` function:

```python

def delete_elements(html, elements=[], delete_before=False, delete_after=False):
    # some code to delete html elements
```
